documentation/en/supraseal.md

```diff
@@ -0,0 +1,264 @@
+
+### Dependencies
+
+CUDA 12.x is required, 11.x won't work.
+
+```bash
+
+The build process depends on GCC 11.x system-wide or gcc-11/g++-11 installed locally.
+* On Arch install https://aur.archlinux.org/packages/gcc11
+* Ubuntu 22.04 has GCC 11.x by default
+* On newer Ubuntu install `gcc-11` and `g++-11` packages
+
+
+### Building
+
+Build the batch-capable Curio binary:
+```bash
+make batch
+```
+
+For calibnet
+```bash
+make batch-calibnet
+```
+
+NOTE: the build should be run on the target machine. Binaries won't be postable between CPU generations due to different
+AVX512 support.
+
+## Configuration
+
+* Run `curio calc batch-cpu` on the target machine to determine supported batch sizes for your CPU
+* Create a new layer configuration for the batch sealer, e.g. batch-machine1:
+
+```toml
+[Subsystems]
+EnableBatchSeal = true
+
+[Seal]
+LayerNVMEDevices = [
+  "0000:88:00.0",
+  "0000:86:00.0", 
+  # Add PCIe addresses for all NVMe devices to use
+]
+
+# Set to your desiced batch size (what the batch-cpu command says your CPU supports AND what you have nvme space for)
+BatchSealBatchSize = 32
+
+# pipelines can be either 1 or 2; 2 pipelines double storage requirements but in correctly balanced systems makes
+# layer hashing run 100% of the time, nearly doubling throughput
+BatchSealPipelines = 2
+
+# Set to true for Zen2 or older CPUs for compatibility
+SingleHasherPerThread = false
+```
+
+### Configure hugepages:
+
+This can be done by adding the following to `/etc/default/grub`. You need 36 1G hugepages for the batch sealer.
+
+```bash
+GRUB_CMDLINE_LINUX_DEFAULT="hugepages=36 default_hugepagesz=1G hugepagesz=1G"
+```
+
+Then run `sudo update-grub` and reboot the machine.
+
+Or at runtime:
+```bash
+sudo sysctl -w vm.nr_hugepages=36
+```
+
+Then check /proc/meminfo to verify the hugepages are available:
+```bash
+cat /proc/meminfo | grep Huge
+```
+
+Expect output like:
+```
+AnonHugePages:         0 kB
+ShmemHugePages:        0 kB
+FileHugePages:         0 kB
+HugePages_Total:      36
+HugePages_Free:       36
+HugePages_Rsvd:        0
+HugePages_Surp:        0
+Hugepagesize:    1048576 kB
+```
+
+Check that `HugePages_Free` is equal to 36, the kernel can sometimes use some of the hugepages for other purposes.
+
+### Setup NVMe devices for SPDK:
+
+NOTE: This is only needed while batch sealing is in beta, future versions of Curio will handle this automatically.
+
+```bash
+cd extern/supra_seal/deps/spdk-v22.09/
+env NRHUGE=36 ./scripts/setup.sh
+```
+
+### PC2 output storage
+
+Attach scratch space storage for PC2 output (batch sealer needs ~70GB per sector in batch - 32GiB for the sealed sector,
+and 36GiB for the cache directory with TreeC/TreeR and aux files)
+
+## Usage
+1. Start the Curio node with the batch sealer layer
+2. Add a batch of CC sectors:
+```bash
+curio seal start --now --cc --count 32 --actor f01234 --layers cluster --duration-days 365
+```
+
+3. Monitor progress - you should see a "Batch..." task running
```

Comment:
Monitor progress:
With a browser pointed to a curio node running --layer=gui you will see 32 sectors with the same task-id on the po-rep views.

Reply:
Added a link to the gui page